cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -749,6 +766,14 @@ public void startSchedule(long basePollSize, long basePollTs, long lastRunSize,
       this.lastRunSize = lastRunSize;
       this.lastRunTs = lastRunTs;
       this.active.set(active);
+      if (persist) {
+        scheduleStore.persist(programId, programType, streamSizeSchedule, basePollSize, basePollTs,
```

Comment:
I am not sure why you persist here. If basePollSize etc. are -1, then there is nothing to persist, and if they are not -1, then that is because you just read from the store. If I understand correctly.